src/argon2/_password_hasher.py

```diff
@@ -16,7 +17,9 @@
 DEFAULT_HASH_LENGTH = RFC_9106_LOW_MEMORY.hash_len
 DEFAULT_TIME_COST = RFC_9106_LOW_MEMORY.time_cost
 DEFAULT_MEMORY_COST = RFC_9106_LOW_MEMORY.memory_cost
-DEFAULT_PARALLELISM = RFC_9106_LOW_MEMORY.parallelism
+DEFAULT_PARALLELISM = (
+    1 if platform.machine() == "wasm32" else RFC_9106_LOW_MEMORY.parallelism
+)
```

Comment:
We could have a more robust check here. I usually like to use

```python
IS_WASM = (sys.platform == "emscripten" or platform.machine() in ["wasm32", "wasm64"])
```

which will extend this to other environments such as WASI or if WASM decides to support 64-bit indices someday.

Reply:
(and similarly for the rest of the places below for this check)

Reply:
Yes, it would be easier to change if the platform will support threading in future